### PR TITLE
Renew cluster slots depends on startNodes (ii)

### PR DESCRIPTION
**Pull request by sazzad16**

### CODE REVIEW

File: src/main/java/redis/clients/jedis/JedisClusterInfoCache.java [hunk 1]
```diff
@@ -162,6 +170,19 @@ public void renewClusterSlots(Jedis jedis) {
             }
           }
         }
+
+        if (startNodes != null) {
```
Comment (yangbodong22011) [inline src/main/java/redis/clients/jedis/JedisClusterInfoCache.java:174]: If `getShuffledNodesPool` exists, it is almost impossible for the code to come here unless all master and slave nodes in the cluster are unserviceable.

File: src/main/java/redis/clients/jedis/JedisClusterInfoCache.java [hunk 2]
```diff
@@ -95,8 +97,14 @@ public JedisClusterInfoCache(final GenericObjectPoolConfig<Jedis> poolConfig,
 
   public JedisClusterInfoCache(final GenericObjectPoolConfig<Jedis> poolConfig,
       final JedisClientConfig clientConfig) {
+    this(poolConfig, clientConfig, null);
+  }
+
+  public JedisClusterInfoCache(final GenericObjectPoolConfig<Jedis> poolConfig,
```
Comment (yangbodong22011) [inline src/main/java/redis/clients/jedis/JedisClusterInfoCache.java:103]: Do you plan to provide an interface for setting `JedisClusterInfoCache` for`JedisCluster`?

Comment (sazzad16) [reply src/main/java/redis/clients/jedis/JedisClusterInfoCache.java:103]: I'm not planning such idea.

But I won't mind to see a PR :)

File: src/main/java/redis/clients/jedis/JedisClusterInfoCache.java (hunk 1 above)
Comment (sazzad16) [reply src/main/java/redis/clients/jedis/JedisClusterInfoCache.java:174]: @yangbodong22011 OK, now I have clearer idea about #2643

**Pull request closed without merging**